conformance/README.md

```diff
@@ -0,0 +1,678 @@
+The outermost clauses `in10`, `in11`, and `in1x` provide the initial set of
+documents to be extended by nested clauses.
+Each *fragment* appends some top-level content to each document in progress.
+Note that at each step, all documents are well-formed.
+
+Fragments are always followed by a _continuation_, which either extends the
+current document(s) with one or more _extension_s, or verifies that they meet an
+expectation, ending that branch of the test tree.
+
+
+These rules describe Ion data-model results for use in the `models` expectation:
+
+```
+model-value   ::=  model-content  |  annotated
+
+model-content ::=  null
+                |  bool
+                |  int
+                |  string
+                |  "("  "null"    model-type    ")"
+                |  "("  "symbol"  model-symtok  ")"
+                |  "("  "list"    model-value*  ")"
+                |  "("  "sexp"    model-value*  ")"
+                |  "("  "struct"  model-field*  ")"
+                |  "("  "blob"    bytes*        ")"
+                |  "("  "clob"    bytes*        ")"
+
+// TODO Other types per denotational semantics
+
+model-symtok  ::=  string
+                |  int
+                |  "("  string  int  ")"  // AbsentSym
+
+model-field   ::=  "("  model-symtok  model-value  ")"
+
+annotated     ::=  "("  "annot"  model-content  string*  ")"
+```
+
+
+# Test Tiers
+
+The full DSL requires a complete Ion text parser at minimum (to read
+the DSL S-expressions), and some aspects require a writer or encoder as well.
+This raises the question of how we can lower the bar so that we can write tests
+that can be exercised for an in-progress Ion implementation earlier than that
+point.
+
+We intend to approach this problem by carefully partitioning the test files,
+and by expressing the earliest phases, that is the lowest-level test cases,
+in JSON. For example:
+
+```
+["in11", ["each",
+            ["text", "1"],
+            ["bytes", 0x51],
+            ["toplevel", 1],
+            ["produces", 1]]]
+```
+
+The next logical tiers would introduce basic symbols and sexps into the parser.
+Here, the `models` expectation becomes valuable:
+
+```
+["in1x", ["text", "(1 2.3)"],
+         ["models", ["sexp", ["int", 1], ["decimal", 23, -1]]]]
+```
+
+The inner clauses should align with the formal data model in the denotational
+semantics, which would (more or less) reduce the components to integers and
+strings.
+
+
+# WIP TODOs
+
+* The denotational test cases are wired with a small catalog of shared symtabs
+  (and soon, shared modules) so that related functionality can be tested.
+  In the stand-alone suite, we should have a similar mechanism.
+* Do we need a "not equivalent" expectation?
+* I’ve got a bunch of test cases (not written in this DSL) that check the bounds
+  of the current symbol table or macro table; that would be nice to have an
+  expectation for.
+  * But that leads to wanting to combine expectations: this input produces this
```

Comment:
As it stands, I don't think it is because the only two expectation clauses cover all possible cases.  But if we add more variety then yeah we'll need ways to express that.  Yesterday I was feeling like `and` and `not` would be useful.  I _think_that gives you non-equivalence.